core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -837,7 +837,10 @@ private[log] class Cleaner(val id: Int,
             logSize + segs.head.size <= maxSize &&
             indexSize + segs.head.offsetIndex.sizeInBytes <= maxIndexSize &&
             timeIndexSize + segs.head.timeIndex.sizeInBytes <= maxIndexSize &&
-            lastOffsetForFirstSegment(segs, firstUncleanableOffset) - group.last.baseOffset <= Int.MaxValue) {
+            //if next Log segment is empty, no need to make sure index relative offset range.
```

Comment:
nit: How about we say: `if first segment size is 0, we don't need to do the index offset range check`.

Reply:
changed as your suggestion